rules/aws_cloudtrail_rules/aws_iam_user_key_created.py

```diff
@@ -6,7 +6,7 @@ def rule(event):
         event.get("eventSource") == "iam.amazonaws.com"
         and event.get("eventName") == "CreateAccessKey"
         and (
-            deep_get(event, "responseElements", "accessKey", "userName")
+            deep_get(event, "responseElements", "accessKey", "userName", default="")
             not in deep_get(event, "userIdentity", "arn")
```

Comment:
This is going to return incorrectly sometimes. `userIdentity.arn` is an arn and `responseElements.accessKey.userName` is the stringified username. 

For the scenario where the responseElement's username is `u` and the `userIdentity.arn` is a IAM user, this would incorrectly return a False

Reply:
I suspect that moving from `not in` to  `useridentity_arn_string.endswith(f"user/{responseelements-username}")` would be accurate more of the time

```python
>>> user = "jack"
>>> userIdentityArn = "arn:::::user/jackson"
>>> user in userIdentityArn
True
>>> userIdentityArn.endswith(f"user/{user}")
False
>>>
```